lesson-4/README.md

```diff
@@ -0,0 +1,46 @@
+# Advanced React - Async
+
+## Goal - understand (unidirectional) data flow
+
+## TODO
+
+* [recap] Component's life cycle hooks recap
+* [recap] State & Props
+* [recap] Children + key (re-render)
+* [recap] Promises + Async
+* [concept] HOC - Higher order components
+* [concept] FAC - Function as children / [Render callback](http://reactpatterns.com/#render-callback)
+* [new] PropTypes - Runtime TypeSafety
+* [new] Effects
+* [new] Local (Components) state vs. Centralized (Atom) state
+* [new] Error handling / Loading (the state machine)
+
+## Problem - "Idea Journal"
+
+* make it work with server - CRUD
+* display loading state indicator
+* validate server payload via propTypes
+
+## Exercises
+
+1. display total number of notes in header // passing state down
+1. prevent saving an empty note (button must disabled)
+1. add propTypes for `Header.js` `Note.js` components
+1. replace the `default` with your github username to ./src/config/api.js`
+1. run `node upload-data.js`
+
+1. implement load notes from server 😈
```

Comment:
is there some exact string of a TODO comment in the code please? this is a bit vague, also not sure how many exercises will be possible to do during 2 hours => for a homework it would need a bit more details I guess..